### ISSUE TITLE
About training

### ISSUE THREAD

**Issue by long123524** [xwjabc/hed#36]
I trained on my own data set, and the following result graph appeared (I am doing a binary classification task). How can I improve it? Please answer
![image](https://user-images.githubusercontent.com/73996375/124350663-7f735700-dc28-11eb-908e-34f44dcb5b81.png)

**Comment by xwjabc** [comment]
It looks a bit weird. Did you use the VGG pre-trained weights?

**Comment by long123524** [comment]
I didn't use it. Do I have to use pre-trained weights to proceed? Because my original image is in 6 bands, adding pre-training weights will report an error

**Comment by xwjabc** [comment]
> I didn't use it. Do I have to use pre-trained weights to proceed? Because my original image is in 6 bands, adding pre-training weights will report an error

Hi @long123524,
For 6 bands, do you mean your input has 6 channels? If that's the case, you can try to load the weights for the VGG backbone except the first layer.

**Comment by long123524** [comment]
![image](https://user-images.githubusercontent.com/73996375/124553375-3bba6080-de67-11eb-8c0d-43ef370a2e83.png)
An error occurred, when load the weights for the VGG backbone except the first layer.
My image input is 6 channels, and I will report an error when I load the pre-training weights

**Comment by long123524** [comment]
An error occurred, when load the weights for the VGG backbone except the first layer.
My image input is 6 channels, and I will report an error when I load the pre-training weights
![image](https://user-images.githubusercontent.com/73996375/124553725-abc8e680-de67-11eb-831a-f8a217e4ea4c.png)

**Comment by long123524** [comment]
I wonder if the first layer of the pre-training weight file should also be changed to 6 channels. But can the pre-training weight file not be modified? Or how to correct

**Comment by long123524** [comment]
I try to load the weights for the VGG backbone except the first layer, however, the result is not very satisfactory.
![image](https://user-images.githubusercontent.com/73996375/124682749-280e0900-defe-11eb-9b2a-475aa22833db.png)
I trained 150 epoch,The picture shows the result of the 150th epoch,What else do I need to do to get good results?

**Comment by long123524** [comment]
I try to load the weights for the VGG backbone include the first layer. The training data is BSDS, and the result is obviously different from my own data set. The VGG weight of the first layer should be more important. How should I modify the weight of the first layer of pre-training so that it can read the 6-channel data set?
![image](https://user-images.githubusercontent.com/73996375/124686959-9a82e700-df06-11eb-88f7-40f012f11700.png)

**Comment by xwjabc** [comment]
Hi @long123524, I wonder if you can try to load the weights for the VGG backbone **excluding** the first layer and train on BSDS dataset? This can help to check if the parameters in the first layer are indeed essential.

**Comment by long123524** [comment]
After removing the first-layer weight of VGG, the BSDS data set will not get good training results or without pre-training VGG, the effect is not good, as shown in the picture.
![image](https://user-images.githubusercontent.com/73996375/124710598-94edc700-df2f-11eb-9fd3-c22e93827a40.png)

Therefore, I think VGG pre-training weights are necessary, and the first layer is also necessary. My idea is to change the first layer of VGG to 6 channels, but how to modify such a pre-training weight file? Or how did this pre-training weight file come from? Can we make a pre-trained VGG ourselves?

**Comment by xwjabc** [comment]
Hi @long123524, the pre-trained weight file comes from [original HED repo](https://github.com/s9xie/hed), and I think this weight file is pretrained on ImageNet classification (may need to double check). It seems the first layer is necessary in your visualization. However, I am not sure if there is a easy way to convert the current 3-channel input layer to 6-channel. In addition, perhaps the 6-channel input of your dataset has different value distribution from the RGB images in HED. Could you tell me what's the meaning of each channel in your data? We can design some conversion given the property of your data.

**Comment by long123524** [comment]
Thank you very much for your reply. My own research direction is remote sensing. The 6 channels I input are the 6 bands in the remote sensing image, and each band has its own spectral information in it.

**Comment by xwjabc** [comment]
Gotcha, I would suggest if you can compute the statistics of the 6 channels (e.g. max/min/std/avg) and choose suitable normalization. Afterward, you can attempt to copy the 3-channel input layer weights twice to make 6-channel input weights and see if this way can help training.

**Comment by long123524** [comment]
![image](https://user-images.githubusercontent.com/73996375/125184462-9cbfab00-e250-11eb-837f-8bf5ea928e49.png)
I successfully got the code to run, but the loss I got was very large, already tens of thousands. Why is this?

**Comment by xwjabc** [comment]
I think you may attempt to lower down the learning rate and try several possible values.
